analyzers/tests/SonarAnalyzer.Test/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Invocation.cs

```diff
@@ -1391,24 +1391,25 @@ public void Invocation_EnumerableCount_SetsNumberConstraint()
                 var x = list.Count();
                 Tag("Before", x);
 
-                x = list.Count(x => true);
-                Tag("Predicate", x);
-
                 list.Clear();
                 x = list.Count();
                 Tag("Clear", x);
 
                 list.Add(42);
                 x = list.Count();
                 Tag("Add", x);
+
+                x = list.Count(_ => true);
+                Tag("Predicate", x);
+
             """;
         var validator = SETestContext.CreateCS(code, "List<int> list").Validator;
         validator.ValidateContainsOperation(OperationKind.Invocation);
 
         validator.TagValue("Before").Should().HaveOnlyConstraints(NumberConstraint.From(0, null), ObjectConstraint.NotNull);
-        validator.TagValue("Predicate").Should().HaveOnlyConstraints(NumberConstraint.From(0, null), ObjectConstraint.NotNull);
         validator.TagValue("Clear").Should().HaveOnlyConstraints(NumberConstraint.From(0), ObjectConstraint.NotNull);
         validator.TagValue("Add").Should().HaveOnlyConstraints(NumberConstraint.From(1, null), ObjectConstraint.NotNull);
+        validator.TagValue("Predicate").Should().HaveOnlyConstraints(ObjectConstraint.NotNull);
```

Comment:
That's unexpected. It should return `NumberConstraint.From(0, null)`. But apparently this is how it was in the old implementation. Two options:
a) Fix it; re-ask for review.
b) Do not fix it and create a hardening ticket or a follow-up PR.